File: Examples/CSharp/Programming-Documents/Joining-Appending/ListKeepSourceFormatting.cs
using System;
using System.IO;

using Aspose.Words;

namespace Aspose.Words.Examples.CSharp.Programming_Documents.Joining_and_Appending
{
    class ListKeepSourceFormatting
    {
        public static void Run()
        {
            //ExStart:ListKeepSourceFormatting
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_JoiningAndAppending();
            string fileName = "TestFile.DestinationList.doc";
            Document dstDoc = new Document(dataDir + fileName);
            Document srcDoc = new Document(dataDir + "TestFile.SourceList.doc");

            // Append the content of the document so it flows continuously.
            srcDoc.FirstSection.PageSetup.SectionStart = SectionStart.Continuous;

            dstDoc.AppendDocument(srcDoc, ImportFormatMode.KeepSourceFormatting);
            dataDir = dataDir + RunExamples.GetOutputFilePath(fileName);
            dstDoc.Save(dataDir);
            //ExEnd:ListKeepSourceFormatting
            Console.WriteLine("\nDocument appended successfully with lists retaining source formatting.\nFile saved at " + dataDir);
        }
    }
}

File: Examples/VisualBasic/Programming-Documents/Joining-Appending/ListKeepSourceFormatting.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Reflection
Imports System.Collections
Imports System.IO
Imports System.Text

Imports Aspose.Words.Lists
Imports Aspose.Words.Fields
Imports Aspose.Words

Public Class ListKeepSourceFormatting
    Public Shared Sub Run()
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_JoiningAndAppending()
        Dim fileName As String = "TestFile.Destination.doc"

        Dim dstDoc As New Document(dataDir & fileName)
        Dim srcDoc As New Document(dataDir & "TestFile.Source.doc")

        ' Append the content of the document so it flows continuously.
        srcDoc.FirstSection.PageSetup.SectionStart = SectionStart.Continuous

        dstDoc.AppendDocument(srcDoc, ImportFormatMode.KeepSourceFormatting)

        dataDir = dataDir & RunExamples.GetOutputFilePath(fileName)
        dstDoc.Save(dataDir)

        Console.WriteLine(vbNewLine & "Document appended successfully with lists keeping source formatting." & vbNewLine & "File saved at " + dataDir)
    End Sub
End Class
